lib/gpx/segment.rb

```diff
@@ -59,8 +59,8 @@ def initialize(opts = {})
     def append_point(pt)
       last_pt = @points[-1]
       if pt.time
-        @earliest_point = pt if(@earliest_point.nil? or pt.time < @earliest_point.time)
-        @latest_point   = pt if(@latest_point.nil? or pt.time > @latest_point.time)
+        @earliest_point = pt if(@earliest_point.nil? or (!@earliest_point.time.nil? and pt.time < @earliest_point.time))
```

Comment:
Suggest removing the `!..nil?` and replacing with a direct `value && other_value` check:

```ruby
(@earliest_point.time && pt.time < @earliest_point.time)
```